src/test/java/com/powsybl/openloadflow/sa/WoodburyDcSecurityAnalysisWithActionsTest.java

```diff
@@ -0,0 +1,454 @@
+/**
+ * Copyright (c) 2024, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.sa;
+
+import com.powsybl.action.Action;
+import com.powsybl.action.GeneratorActionBuilder;
+import com.powsybl.action.PhaseTapChangerTapPositionAction;
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.contingency.BranchContingency;
+import com.powsybl.contingency.Contingency;
+import com.powsybl.contingency.ContingencyContext;
+import com.powsybl.contingency.LoadContingency;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.serde.test.MetrixTutorialSixBusesFactory;
+import com.powsybl.loadflow.LoadFlowParameters;
+import com.powsybl.openloadflow.OpenLoadFlowParameters;
+import com.powsybl.openloadflow.network.PhaseControlFactory;
+import com.powsybl.openloadflow.util.LoadFlowAssert;
+import com.powsybl.security.SecurityAnalysisParameters;
+import com.powsybl.security.SecurityAnalysisResult;
+import com.powsybl.security.condition.TrueCondition;
+import com.powsybl.security.monitor.StateMonitor;
+import com.powsybl.security.results.BranchResult;
+import com.powsybl.security.results.OperatorStrategyResult;
+import com.powsybl.security.strategy.OperatorStrategy;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.concurrent.CompletionException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Pierre Arvy {@literal <pierre.arvy at artelys.com>}
+ */
+class WoodburyDcSecurityAnalysisWithActionsTest extends AbstractOpenSecurityAnalysisTest {
+
+    private LoadFlowParameters parameters;
+    private SecurityAnalysisParameters securityAnalysisParameters;
+
+    @BeforeEach
+    public void setUpWoodburyDcSa() {
+        securityAnalysisParameters = new SecurityAnalysisParameters();
+        // configure sa to use Woodbury dc sa
+        parameters = new LoadFlowParameters();
+        parameters.setDc(true);
+        securityAnalysisParameters.setLoadFlowParameters(parameters);
+        OpenSecurityAnalysisParameters openSecurityAnalysisParameters = new OpenSecurityAnalysisParameters();
+        openSecurityAnalysisParameters.setDcFastMode(true);
+        securityAnalysisParameters.addExtension(OpenSecurityAnalysisParameters.class, openSecurityAnalysisParameters);
+    }
+
+    @Test
+    void testSaDcPhaseTapChangerTapPositionAction() {
+        Network network = MetrixTutorialSixBusesFactory.create();
```

Comment:
This looks very much like testSaDcPhaseTapChangerTapPositionAction
Can you, instead of duplicating the test, turn testSaDcPhaseTapChangerTapPositionAction into a parametrized test that would run in fast mode and in default mode ?